src/test/osd/TestPGLog.cc

```diff
@@ -2064,6 +2066,269 @@ TEST_F(PGLogTest, ErrorNotIndexedByObject) {
   EXPECT_EQ(del.reqid, entry->reqid);
 }
 
+class PGLogMergeDupsTest : public ::testing::Test, protected PGLog {
```

Comment:
looks like you've got extensive tests of merging - it'd be good to have some for the trimming behavior as well